torchaudio/transforms.py

```diff
@@ -1025,3 +1025,127 @@ def forward(self, waveform: Tensor) -> Tensor:
             hp_lifter_freq=self.hp_lifter_freq,
             lp_lifter_freq=self.lp_lifter_freq,
         )
+
+
+class Synth(torch.nn.Module):
+    r"""This effect can be used to generate fixed or swept frequency audio tones with various wave shapes,
+     or to generate noise.
+
+    Args:
+        sample_rate (int, optional): Sample rate of audio signal (Default: ``16000``).
+        duration (float, optional): Duration in seconds of audio to synthesise (Default: ``1.0``).
+        wave_type (str, optional): is one of sine, square, triangle, sawtooth, trapezium, exp, white noise
+         and brownian noise (Default: ``"sine"``).
+        freq (int or tuple, optional): are the frequencies at the beginning/end of synthesis in Hz (Default: ``440``).
+        chirp (str, optional): If `freq` defined the variation between to frequencies can be changed `linear`,
+            `square` or `exp` (Default: ```linear``).
+        amp (float, optional): Maximum amplitude (Default: ```1.0``).
+    """
+
+    def __init__(self,
```

Comment:
The amount of state carried around seems pretty light, further most of these arguments don't apply to all methods and also the functions could be combined to yield the usecases.

For example we could do:
```sine(linear_freqs(start=50, end=500)(phases(sample_rate=16000, duration=1.0)))```

I think phases and linear_freqs might even be useful as standalone transforms anyway maybe even for other methods. We already have functions that return frequencies for e.g. melspectrogram. We could potentially use this to create transforms with different interesting filters, which is unexpectely beyond the scope of synthesization.